src/crimson/osd/main.cc

```diff
@@ -66,6 +68,35 @@ auto partition_args(seastar::app_template& app, char** argv_begin, char** argv_e
   return make_pair(std::move(ceph_args), std::move(app_args));
 }
 
+bool set_crimson_daemon(std::vector<const char*>& ceph_args, Preforker& forker){
+  bool crimson_daemonize = true;
+  for (std::vector<const char*>::iterator i = ceph_args.begin(); i != ceph_args.end();++i) {
+    if (ceph_argparse_flag(ceph_args, i, "--foreground", "-f", (char*)NULL)){
+      crimson_daemonize = false;
+      break;
+    }
+  }
+ 
+  std::string err_msg;
+  int err = forker.prefork(err_msg);
+  if (err < 0) {
+      cerr << err_msg << std::endl;
+      forker.exit(err);
+  }
+  if (forker.is_parent()) {
+      int err = forker.parent_wait(err_msg);
+      if (err < 0) {
+        cerr << err_msg <<std::endl;
+      }
+      forker.exit(err);
+  }
+  setsid();
+  if (crimson_daemonize)
+    forker.daemonize();
+
+  return crimson_daemonize;
```

Comment:
no need to memorize this setting. please check `Preforker::signal_exit()` the more details.